src/protocols/internal/encode.c

```diff
@@ -103,18 +105,23 @@ static ssize_t internal_encode(fr_dbuff_t *dbuff,
 
 	/*
 	 *	Leave one byte in hopes that the length will fit
-	 *	and remember where the encoded data starts.
+	 *	so we needn't move the encoded data.
 	 */
 	fr_dbuff_marker(&len_field, &work_dbuff);
 	FR_DBUFF_ADVANCE_RETURN(&work_dbuff, 1);
-	fr_dbuff_marker(&value_field, &work_dbuff);
-	fr_dbuff_marker(&value_dest, &work_dbuff);
+
+	/*
+	 *	Create dbuff to hold encoded data that assures space
+	 *	for the length.
+	 */
+	value_dbuff = FR_DBUFF_RESERVE(&work_dbuff, sizeof(uint64_t) - 1);
+	fr_dbuff_marker(&value_field, &value_dbuff);
 
 	switch (da->type) {
 	case FR_TYPE_VALUE:
-		slen = fr_value_box_to_network(&FR_DBUFF_RESERVE(&work_dbuff, sizeof(uint64_t) - 1), &vp->data);
+		slen = fr_value_box_to_network(&value_dbuff, &vp->data);
 		if (slen < 0) return PAIR_ENCODE_FATAL_ERROR;
-		FR_PROTO_HEX_DUMP(fr_dbuff_current(&value_field), slen, "value %s",
+		FR_PROTO_HEX_DUMP(fr_dbuff_start(&value_dbuff), slen, "value %s",
```

Comment:
Mmm, should that be start? Probably actually need a marker before calling `fr_value_box_to_network` i.e. what was there before :)